library/std/src/sync/condvar.rs

```diff
@@ -195,8 +195,10 @@ impl Condvar {
         if poisoned { Err(PoisonError::new(guard)) } else { Ok(guard) }
     }
 
-    /// Blocks the current thread until this condition variable receives a
-    /// notification and the provided condition is false.
+    /// Blocks the current thread until the provided `condition` is met.
+    ///
+    /// While `condition` is not met, this function will [`wait`] for
+    /// notification, before again checking `condition`.
```

Comment:
Maybe we could make it pretty explicit about the order things happen? Something like

```rust
    /// Blocks the current thread until the provided condition becomes `false`.
    ///
    /// `condition` is checked immediately; if not met (returns `true`), this will [`wait`]
    /// for the next notification then check again. This repeats until `condition` returns
    /// `false`, in which case this function returns.
```

(also dropped backticks from the summary condition, since I think this is referring to the parameter by quality rather than by name)

Reply:
I like it! Very clear. I was leaning toward the same, but realized there might have been reasons not to let the documented interface completely reflect the implementation.

But I agree it's a much more useful and clear way to define the semantics. :+1:

Reply:
I think the behavior is relevant enough here that if we had to change it for whatever reason, it would surprise enough people that we would also need to change the docs 🙂 I don't think this is too likely though.

Feel free to reword to something similar to the above as you see fit then squash and push, and I'd be happy to merge this.

Reply:
Tweaked and rebased :+1: